theano/tensor/basic.py

```diff
@@ -7138,11 +7135,15 @@ def make_node(self, a, choices):
                 "choose first argument must have an [u]int* dtype. Got %s." % a.dtype
             )
 
+        # Only use make_list if choices have inconsistent shapes
+        # otherwise use as_tensor_variable
+        choice_ndim = None
         if isinstance(choices, (tuple, list, theano.typed_list.TypedListVariable)):
-            choice = theano.typed_list.make_list(choices)
-            choice_ndim = choice.ttype.ndim
-            choice_bcast = choice.ttype.broadcastable
-        else:
+            if builtins.any(c.ndim > 1 for c in choices):
```

Comment:
It looks like we could combine these two conditions into a single "do/don't use `make_list`" `if`-`else` block.  If so, this situation would be much more concise and we wouldn't need to set `choice_ndim = None` and use it later in another `if` statement.

Reply:
Using this approach will return an error if `choices` is not iterable when running `builtins.any(c.ndim > 1 for c in choices)`, that is why the type is estimated first here. I agree that the `choice_ndim = None` trick makes things more complex but I did not find another shorter way to test that.

Reply:
Ah, if that's happening, then it's probably due to the `theano.typed_list.TypedListVariable`, since the other two cases in the `isinstance` are definitely iterable.  That would also imply that we're never actually seeing the case in which `choices` is a `TypedListVariable`, because, if the first condition passes, then we're necessarily going to check the second condition and fail for the same reasons.

Something like this should work:
```python
        if (
            isinstance(choices, (tuple, list))
            and builtins.any(c.ndim > 1 for c in choices)
        ) or isinstance(choices, theano.typed_list.TypedListVariable):
            choice = theano.typed_list.make_list(choices)
            choice_ndim = choice.ttype.ndim
            choice_bcast = choice.ttype.broadcastable
        else:
            choice = as_tensor_variable(choices)
            choice_ndim = choice.ndim - 1
            choice_bcast = choice.broadcastable[1:]
```

Reply:
This has been changed on `basic.py`.

Should I also add a test condition for `TypedListVariable`? In either case, we can probably remove the import line [7130](https://github.com/pymc-devs/Theano-PyMC/blob/4249a01c886d034abf5a29bb40c2a0d06c56106e/theano/tensor/basic.py#L7133) as it seems unused here.

Reply:
Yes, adding one would be a long overdue improvement!  There definitely should've been tests for `TypedListVariable`s (e.g. both passed to the `Choose` `Op` and converted to one within `Choose.make_node`), even if they would've been an expected failure.  

I believe the import line is there because `theano.typed_list` isn't automatically imported, so the lines that use `theano.typed_list.TypedListVariable` and `theano.typed_list.make_list` would fail.  If that's not the case, then we can definitely remove it.